src/sidebar/build-thread.js

```diff
@@ -203,96 +198,93 @@ function sort(threads, compareFn) {
 }
 
 /**
- * Return a copy of `thread` with siblings of the top-level thread sorted according
- * to `compareFn` and replies sorted by `replyCompareFn`.
+ * Return a new `Thread` object with all (recursive) `children` arrays sorted.
+ * Sort the children of top-level threads using `compareFn` and all other
+ * children using `replyCompareFn`.
+ *
+ * @param {Thread} thread
+ * @param {(a: Annotation, b: Annotation) => boolean} compareFn - Less-than
+ *         comparison function for sorting top-level annotations
+ * @param {(a: Annotation, b: Annotation) => boolean} replyCompareFn - Less-than
+ *       comparison function for sorting replies
+ * @return {Thread}
  */
 function sortThread(thread, compareFn, replyCompareFn) {
-  const children = thread.children.map(function (child) {
-    return sortThread(child, replyCompareFn, replyCompareFn);
-  });
+  const children = thread.children.map(child =>
+    sortThread(child, replyCompareFn, replyCompareFn)
+  );
 
-  return Object.assign({}, thread, {
-    children: sort(children, compareFn),
-  });
+  return { ...thread, ...{ children: sort(children, compareFn) } };
 }
 
 /**
- * Return a copy of @p thread with the `replyCount` and `depth` properties
+ * Return a copy of thread with the `replyCount` and `depth` properties
```

Comment:
```suggestion
 * Return a copy of `thread` with the `replyCount` and `depth` properties
```

Make it a bit clearer that `thread` refers to a parameter here.